src/test/java/pages/PlayListPage.java

```diff
@@ -0,0 +1,328 @@
+        new Actions(driver)
+                .doubleClick(playListB)
+                .perform();
+    }
+
+    public void selectPLName() throws InterruptedException {
+        WebElement playLInputField = driver.findElement(By.cssSelector("input[name='name'] "));
+        Thread.sleep(2000);
+        playLInputField.sendKeys(Keys.CONTROL, "a", Keys.BACK_SPACE);
+        Thread.sleep(1000);
+        playLInputField.sendKeys("King41");
+        playLInputField.sendKeys(Keys.ENTER);
+        Thread.sleep(1000);
+    }
+
+    public boolean isDisplayedNewPL() {
+        WebElement newPlayListName = driver.findElement(By.xpath("//li[@class='playlist playlist'][4]"));
+        Assert.assertTrue(newPlayListName.isDisplayed());
+        return true;
+    }
+
+    // Play songs test
+    public void clickAllSongsBtn() {
+        WebElement allSongsBtn = driver.findElement(By.xpath("//nav[@id='sidebar']/section[@class='music']//a[@href='#!/songs']"));
+        allSongsBtn.click();
+    }
+
+    public void doubleClickSongBtn() {
+        WebElement songBtn = driver.findElement(By.cssSelector(".all-songs [draggable='true']:nth-of-type(6) .title "));
+        new Actions(driver)
+                .doubleClick(songBtn)
+                .perform();
+    }
+
+    public boolean asssertEqualizerPlaying() {
+        WebElement equalizerBtn = driver.findElement(By.cssSelector("[alt='Sound bars']"));
+        Assert.assertTrue(equalizerBtn.isDisplayed());
+        return true;
+    }
+
+    public void createPlayList(String plName) {
+        // click create playList
+        WebElement createPlBtn = driver.findElement(By.xpath("//section[@id='playlists']//i[@role='button'] "));
+        createPlBtn.click();
+        // new PLBtn
+        WebElement newPLBtn = driver.findElement(By.cssSelector("[data-testid='playlist-context-menu-create-simple'] "));
+        newPLBtn.click();
+
+        // enter PL name
+        WebElement pLNameField = driver.findElement(By.xpath("//section[@id='playlists']//input[@name='name'] "));
+        pLNameField.click();
+        pLNameField.clear();
+        waitUntilVisible(pLNameField);
+        pLNameField.sendKeys("Rock Stars 5");
+        pLNameField.sendKeys(Keys.ENTER);
+        waitUntilVisible(driver.findElement(By.xpath("//*[text()='" + plName + "']")));
+
+    }
+
+    public List<String> getplayListNames() {
+        List<String> playListNames = new ArrayList<>();
+        List<WebElement> playLists = waitForElementsVisibility(driver.findElements(By.xpath("//li[@class='playlist playlist']")));
+
+        for (WebElement playList : playLists) {
+            playListNames.add(playList.getText());
+        }
+        return playListNames;
+
+    }
+
+    // play songs in Albums
+    public void clickAlbumsBtn() throws InterruptedException {
+        WebElement albumBtn = driver.findElement(By.xpath("//nav[@id='sidebar']/section[@class='music']//a[@href='#!/albums']"));
+        Thread.sleep(1000);
```

Comment:
Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.

![image](https://github.com/Test-Pro-Bootcamp/A42-QA-Automation-Framework/assets/104761482/4882e9d4-47fd-48eb-b65f-0bfcfaad8a02)